File: C++/1021.cpp
 __________________________________________________________________________________________________
sample 4 ms submission
class Solution {
public:
    string removeOuterParentheses(string S) {        
        int num_left = 0;
        int num_right = 0;
        int break_point = 0;
        string ret = "";
        for (int idx = 0; idx < S.length(); ++idx) {
            if (S[idx] == '(') {
                num_left++;
            }
            else {
                num_right++;
            }
            if (num_left == num_right) {
                ret.append(S.substr(break_point+1, idx-break_point-1));
                num_left = num_right = 0;
                break_point = idx+1;
            }
        }
        return ret;
    }
};
__________________________________________________________________________________________________
sample 8 ms submission
class Solution {
public:
    string removeOuterParentheses(string S) {
        int n = S.length();
        int i = 0,j = 0;
        int a = 0;
        string ans = "";
        while(j < n){
            if(S[j] == '('){
                a -= 1;
            }
            else if(S[j] == ')'){
                a += 1;
            }
            if(a == 0 && j > 0){
                int k = i+1;
                while(k<j){
                    ans += S[k];
                    k++;
                }
                i = j + 1;
            }
            j++;
        }
        return ans;
    }
};
__________________________________________________________________________________________________
class Solution {
public:
     string removeOuterParentheses(string S) {
        string res;
        int opened = 0;
        for (char c : S) {
            if (c == '(' && opened++ > 0) res += c;
            if (c == ')' && opened-- > 1) res += c;
        }
        return res;
    }
};

File: Python3/1021.py
__________________________________________________________________________________________________
sample 24 ms submission
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        count = 0
        res = ''
        v = 0
        for i in range(len(S)):
            if S[i] == '(':
                count += 1
            elif S[i] == ')':
                count -= 1
                if count == 0:
                    res += S[v + 1 : i]
                    v = i + 1
        return res
__________________________________________________________________________________________________
sample 13016 kb submission
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        
        outter = 0
        x = ''

        for i in S:
            if i == '(' and outter > 0:
                x += i
                outter += 1
            if i == ')' and outter > 1:
                x += i
                outter -= 1
            outter += 1 if i == '(' else -1


        return x
__________________________________________________________________________________________________
